src/Query/Builder.php

```diff
@@ -966,8 +967,12 @@ public function where($column, $operator = null, $value = null, $boolean = 'and'
             }
         }
 
-        if (func_num_args() === 1 && is_string($column)) {
-            throw new ArgumentCountError(sprintf('Too few arguments to function %s("%s"), 1 passed and at least 2 expected when the 1st is a string.', __METHOD__, $column));
+        if (func_num_args() === 1 && ! is_array($column) && ! is_callable($column)) {
+            throw new ArgumentCountError(sprintf('Too few arguments to function %s(%s), 1 passed and at least 2 expected when the 1st is not an array or a callable', __METHOD__, var_export($column, true)));
+        }
+
+        if (! is_int($column) && ! is_string($column)) {
+            throw new InvalidArgumentException(sprintf('First argument of %s must be a column name as "string". Got "%s"', __METHOD__, get_debug_type($column)));
```

Comment:
I assume you intentionally do not want to tell people that an "int" is accepted, since you silently convert it. SGTM.